### ISSUE TITLE
More detailed output for assertion failures

### ISSUE THREAD

**Issue by ephja** [nim-lang/Nim#1977]
unittest.check outputs the value of each sub-expression in the predicate upon failure, but it depends on other things in the unittest module.

**Comment by dom96** [comment]
I think the output should also be improved. It's a tad hard to read at the minute.

**Comment by Varriount** [comment]
Exactly what kind of output is wanted? I assume that this is for the standard 'assert' template, not anything in the unittest module?

**Comment by dom96** [comment]
No, the output for ``check``.

**Comment by jlp765** [comment]
One improvement would be showing what didn't match, especially for procedure calls,eg

Currently (version 0.10.2): 
Error: unhandled exception: myProc("some string") == "whatItShouldBe"  [AssertionError]
Error: execution of an external program failed

Would be nice if it showed what was compared (eg, include the output of the procedure call)
something like adding the extra middle line:

Error: unhandled exception: myProc("some string") == "whatItShouldBe"  [AssertionError]
Error: Assert failed:  "firstStringResult" == "whatItShouldBe"
Error: execution of an external program failed

**Comment by Varriount** [comment]
The problem with this is that the comparison cannot always be represented using textual information, not to mention the additional risk of errors when converting the given information to a string representation.

**Comment by ephja** [comment]
@Varriount That's true, but it might still be helpful. The representation could be optional and it could be printed after the usual assertion failure message.

**Comment by superfunc** [comment]
This one can probably be closed, as of 0.17.2, the output for this test:
```nim
import unittest

proc foo(i: int, j: int) : int =
    if i == 3:
        return 4
    else:
        return j

suite "description for this stuff":
  test "testy":
    check(foo(5, 5) == 4)
```

is:
```bash
[Suite] description for this stuff
    ut.nim(11, 20): Check failed: foo(5, 5) == 4
    foo(5, 5) was 5
  [FAILED] testy
Error: execution of an external program failed:
```

**Comment by johnnovak** [comment]
That's a good improvement 👍
